fxevent/zap.go

```diff
@@ -71,5 +71,9 @@ func (l *ZapLogger) LogEvent(event Event) {
 		l.Logger.Error("startup failed, rolling back", zap.Error(e.StartErr))
 	case *Running:
 		l.Logger.Info("running")
+	case *LoggerError:
+		l.Logger.Error("error constructing logger, did not get back fxevent.Logger from WithLogger", zap.Error(e.Err))
```

Comment:
nit: why we were unable to build the logger should be in the error message; the extra string can be skipped
```suggestion
		l.Logger.Error("error constructing logger", zap.Error(e.Err))
```